content/protobuf/index.md

```diff
@@ -1,7 +1,120 @@
 ---
-title: Protobuf Overview
-showAside: false
+title: Protobuf
+description: Using Protocol Buffers with IceRPC
 showReadingTime: false
 ---
 
-This is a placeholder for the Protobuf Overview page.
+## About Protocol Buffers
+
+Protocol Buffers or Protobuf for short is a popular [Interface Definition Language] (IDL) and serialization format
+created by Google and available for many programming languages. [gRPC] and several other RPC frameworks use [Protobuf] as
+their IDL.
+
+This chapter describes the IceRPC + Protobuf integration, not Protobuf itself. The IceRPC + Protobuf integration uses
+standard Protobuf files, with the regular Protobuf syntax and semantics.
+
+## The IceRPC + Protobuf integration
+
+The IceRPC + Protobuf integration helps you implement Protobuf services with IceRPC and call such services using (over)
+IceRPC.
+
+In practical terms, the IceRPC + Protobuf integration for C#:
+
+- defines an IceRPC-specific client-side and server-side mapping for Protobuf services to C#
+- provides a `protoc` plugin (`protoc-gen-icerpc-csharp`) that generates code for Protobuf services
+- provides a support assembly used by this generated code, `IceRpc.Protobuf.dll`
+- provides a MSBuild integration that helps you execute `protoc` with the `protoc-gen-icerpc-csharp` plugin in MSBuild
+projects: `IceRpc.Protobuf.Tools`
+
+User-defined Protobuf types such as messages and enums are handled by the `protoc` compiler and its built-in C# plugin.
+The IceRPC + Protobuf integration does not customize this mapping in any way.
+
+The net result is you need to write very little code to call or implement Protobuf services with IceRPC.
+
+For example, let's take a typical Greeter example:
+
+```protobuf
+syntax = "proto3";
+
+package visitor_center;
+option csharp_namespace = "VisitorCenter";
+
+// A simple Greeter service.
+service Greeter {
+    // Creates a personalized greeting.
+    rpc Greet (GreetRequest) returns (GreetResponse);
+}
+
+// The request contains the name of the person to greet.
+message GreetRequest {
+    string name = 1;
+}
+
+// The response contains the greeting.
+message GreetResponse {
+    string greeting = 1;
+}
+```
+
+`protoc`'s built-in plugin for C# generates C# classes for the `GreetRequest` and `GreetResponse` messages, while
+`protoc-gen-icerpc-csharp` generates two C# interfaces and a struct for the `Greeter` service:
```

Comment:
I don't think it's necessary to qualify `classes` and `interfaces` as C# classes and C# interfaces. It's already pretty clear from the context I think.

Or, we should at least be consistent and also qualify 'struct' (in the last line) as `C# struct`